From a722d933f64960270c6acd38c112ae9d5730e345 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 24 Oct 2018 15:26:20 +0800
Subject: [PATCH 5004/5242] MLK-20046-1: ISI: correct RGB24 and BGR24 format
 configuration

commit  9121d3816c4432641200e6c74fe939fd1774c4aa from
https://source.codeaurora.org/external/imx/linux-imx.git

According to V4L2 doc, RGB24 and BGR24 should have three bytes
for each pixel rather than four, so correct it.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit 3694f7ffe2443d86ae740f12393edc007c997a98)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-cap.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-cap.c b/drivers/media/platform/imx8/mxc-isi-cap.c
index 564fdef..703cfed 100644
--- a/drivers/media/platform/imx8/mxc-isi-cap.c
+++ b/drivers/media/platform/imx8/mxc-isi-cap.c
@@ -45,8 +45,8 @@ struct mxc_isi_fmt mxc_isi_out_formats[] = {
 	}, {
 		.name		= "RGB24",
 		.fourcc		= V4L2_PIX_FMT_RGB24,
-		.depth		= { 32 },
-		.color		= MXC_ISI_OUT_FMT_XRGB32,
+		.depth		= { 24 },
+		.color		= MXC_ISI_OUT_FMT_RGB32P,
 		.memplanes	= 1,
 		.colplanes	= 1,
 		.mbus_code  = MEDIA_BUS_FMT_RGB888_1X24,
@@ -61,8 +61,8 @@ struct mxc_isi_fmt mxc_isi_out_formats[] = {
 	}, {
 		.name		= "BGR24",
 		.fourcc		= V4L2_PIX_FMT_BGR24,
-		.depth		= { 32 },
-		.color		= MXC_ISI_OUT_FMT_XBGR32,
+		.depth		= { 24 },
+		.color		= MXC_ISI_OUT_FMT_BGR32P,
 		.memplanes	= 1,
 		.colplanes	= 1,
 		.mbus_code  = MEDIA_BUS_FMT_BGR888_1X24,
-- 
1.7.9.5

